python/cudf/cudf/core/series.py

```diff
@@ -815,6 +817,100 @@ def __array_function__(self, func, types, args, kwargs):
         else:
             return NotImplemented
 
+    def map(self, arg, na_action=None) -> "Series":
+        """
+        Map values of Series according to input correspondence.
+
+        Used for substituting each value in a Series with another value,
+        that may be derived from a function, a ``dict`` or
+        a :class:`Series`.
+
+        Parameters
+        ----------
+        arg : function, collections.abc.Mapping subclass or Series
+            Mapping correspondence.
+        na_action : {None, 'ignore'}, default None
+            If 'ignore', propagate NaN values, without passing them to the
+            mapping correspondence.
+
+        Returns
+        -------
+        Series
+            Same index as caller.
+
+        Examples
+        --------
+        >>> s = cudf.Series(['cat', 'dog', np.nan, 'rabbit'])
+        >>> s
+        0      cat
+        1      dog
+        2      NaN
+        3   rabbit
+        dtype: object
+
+        ``map`` accepts a ``dict`` or a ``Series``. Values that are not found
+        in the ``dict`` are converted to ``NaN``, unless the dict has a default
+        value (e.g. ``defaultdict``):
+
+        >>> s.map({'cat': 'kitten', 'dog': 'puppy'})
+        0   kitten
+        1    puppy
+        2      NaN
+        3      NaN
+        dtype: object
+
+        It also accepts numeric functions:
+        >>> s = cudf.Series([1, 2, 3, 4, np.nan])
+        >>> s.map(lambda x: x ** 2)
+        0       1
+        1       4
+        2       9
+        3       16
+        4       NaN
+        dtype: int64
+
+        Please note map currently only supports fixed-width numeric types.
```

Comment:
This line is a bit confusing to me: do you mean the lambda function only supports numeric->numeric mapping?

Reply:
if means that map only supports numeric type callables. It does not support string or categorical type callables as pandas does. (We wanted to make it as broad as possible so did not specify the second part.)

Reply:
I changed this statement to 'Please note map currently only supports fixed-width numeric type functions.' Hopefully, that helps make it a bit clearer :D